classes/auth/login.py

```diff
@@ -0,0 +1,113 @@
+            name: user details
+            description: User's email and password
+            type: string
+            required: true
+            schema:
+              id: login
+              properties:
+                email:
+                  default: jimmy@gmail.com
+                password:
+                  default: pass1234
+        responses:
+          200:
+            description: Login Successful
+          400:
+            description: Invalid email
+          401:
+            description: Invalid access token
+          404:
+            description: User not registered!
+          412:
+            description: The password is too short
+          422:
+            description: Please fill all the fields
+        """
+        email_pattern = r"([a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
+        user_email = str(request.data.get('email', '')).strip().lower()
+        user_password = str(request.data.get('password', ''))
+        user = Users.query.filter_by(email=user_email, password=user_password).first()
+
+        if not user_email and not user_password:
+            return make_response(jsonify({'message': 'Please fill all the fields'})), 400
+
+        if not re.search(email_pattern, user_email):
+            return make_response(jsonify({'message': 'Invalid email given'})), 400
+
+        if len(user_password) < 7:
+            return make_response(jsonify({'message': 'The password is too short'})), 412
+
+        if not user:
+            return make_response(jsonify({'message': 'User not registered!'})), 404
+
+        access_token = jwt.encode({'id': user.id, 'expiry_time': str(datetime.datetime.utcnow() +
+                                                                     datetime.timedelta(minutes=30))},
+                                  os.getenv('SECRET', '$#%^%$^%@@@@@56634@@@'))
+
+        if access_token:
+            return make_response(jsonify({'access_token': access_token.decode(),
+                                          'message': 'Successful login'})), 200
+        else:
+
+            return make_response(jsonify({'message': 'Invalid access token'})), 401
+
+
+class UserLogoutAuthentication(MethodView):
+    """This will enable user to destroy the session of the current user.
+    """
+    decorators = [token_required]
+
+    @classmethod
+    def post(self, user_in_session):
+        """Method to logout the user
+
+         User logout
+        ---
+        tags:
+          - Authentication
+        responses:
+          200:
+            description: User logged out successfully
+          401:
+            description: Invalid access token
+          409:
+            description: The user is already logged out!
+        """
+        access_token = request.headers.get('x-access-token')
+        if access_token:
+            user = Users.query.filter_by(id=user_in_session).first()
+            save_tokens = BlacklistToken(token=access_token)
+            save_tokens.save()
```

Comment:
Why are you saving the token to the db at this point? I thought according to your model implementation it should only be saved if the user has logged out before the expiry of the token.